Training/Training/Mealy.c

```diff
@@ -0,0 +1,93 @@
+#define _CRT_SECURE_NO_WARNINGS
+#include <stdio.h>
+#include <string.h>
+#include <windows.h>
+#include <io.h>
+
+/// <summary> states of the Mealy machine
+/// A state transition diagram is attached for clear understanding </summary>
+typedef enum {
+   S0,  // Initial state
+   S1,  // After '0'
+   S2,  // After '01'
+   S3,  // After '011'
+   A1,  // After '1'
+   A2,  // After '11'
+   A3,  // After '110'
+} State;
+
+void Mealy (char inpPath[], char outPath[]);
+State NextMealyState (State currentState, int input, int* output);
+
+void main (int argc, char* argv[]) {
+   Mealy (argv[1], argv[2]);
+}
+
+// Function to open input file and create a new file and store the result
+void Mealy (char inpPath[], char outPath[]) {
+   FILE* inpTestFile = fopen (inpPath, "r"), * outTestFile = fopen (outPath, "w");
+   if (inpTestFile == NULL || outTestFile == NULL) {
+      printf ("Error opening file");
+      return;
+   }
+   State currentState = S0;
+   int output = 0, i = 0, inpFileSize = filelength (fileno (inpTestFile)) + 1;
+   char* inpFileString = (char*)malloc (inpFileSize * sizeof (char));

```

Comment:
After any memory allocation call, you should check if the allocation succeeded. Here you are making an assumption that the allocation succeeded. Although it will not be problem for tutorial programs like this, you should learn and use the techniques of defensive programming.